### ISSUE TITLE
New user registration emails not going through?

### ISSUE THREAD

**Issue by brashrebel** [realbig/rbw-functionality#6]
I just tried to signup a new user (missy) and she didn't get a new user email to activate her account. I manually activated the account and then sent her a request to join another site (roofing) and she didn't receive this either.

Don't have time to test more but we really need to ensure that emails are all working properly.

**Comment by d4mation** [comment]
At first I thought that maybe this could have been the result of our server configuration changing behind-the-scenes, so I hooked into `admin_init` in our Multisite plugin to send a test email and I received it successfully.

```
add_action( 'admin_init', function() {
    $email = 'eric@realbigmarketing.com';
    $fromemail = 'welcome@realbigwebsites.com';
    $subject = 'email test';
    $msg = 'This is a test';
    $headers = "From: ".$fromemail;
    if( ! mail( $email, $subject, $msg, $headers ) ) {
        echo "Problem sending email.";
    }
    else{
        echo "Email successfully sent";
    }
} );
```

![image](https://cloud.githubusercontent.com/assets/7770631/14715603/64c8bdb4-07b7-11e6-95f4-ff0e771002a2.png)

WordPress uses PHP's `mail()` function whenever it sends an email&mdash;It seems like that much is working.

I have encountered issues in the past where Users have not received emails due to filters in place by their company/email provider that checks to see if the `FROM` header includes a real email address and rejects it if it is not (Sites using the default configuration of `wordpress@<siteurl>` all got rejected). [Since ours isn't real, it is technically being sent via `wpengine.com`.](https://support.google.com/mail/answer/1311182?hl=en)

Unless there's something in particular blocking User Registration functions from completing and sending an email (and not outputting some kind of error notice), it would seem that it should be working.

I'm going to test this with a fresh email account and see if it gets the email. If not, I'll try to troubleshoot this some more.

**Comment by d4mation** [comment]
I just created a new User Account (With a Gmail Account, so it should not have been filtered) and it did not receive an email either. 

Going to see what I can do to debug this. Very strange stuff.

**Comment by d4mation** [comment]
I ended up contacting WP Engine to try to rule out the server since there aren't many Actions/Filters around the User Registration functions that would be useful to try to determine if the mailing function failing.

Strangely enough, the support tech was able to successfully receive an email at both the network level and the sub-site level. He brought in a senior tech as well and neither of them were able to reproduce this bug. They were using `@wpengine.com` email addresses when they were testing.

They recommended trying to use a plugin like [WP SMTP](https://wordpress.org/plugins/wp-smtp/) to try to rule out the server as much as possible. 

It's a really weird situation. Their email addresses may have worked because the domain is whitelisted by the server, but there may be no good way to determine that without doing something more drastic like rerouting all our mailing requests to SMTP as they suggested.

Where do you think we should go from here? Switching to SMTP is drastic, but it may work.

**Comment by BigActual** [comment]
Is it possible that the sending address is black listed?

**Comment by brashrebel** [comment]
Yeah, we need to setup some SMTP integration. I'll open up a new issue.
